src/data/complex/exponential.lean

```diff
@@ -1510,6 +1526,75 @@ calc cos 2 = cos (2 * 1) : congr_arg cos (mul_one _).symm
           zero_le_two) _
   ... < 0 : by norm_num
 
+lemma exp_bound_div_one_sub_of_interval_approx  {x : ℝ} (h1 : 0 ≤ x) (h2 : x ≤ 1) :
+  ∑ (j : ℕ) in finset.range 3, x ^ j / (j.factorial)
+  + x ^ 3 * ((3 : ℕ) + 1) / ((3 : ℕ).factorial * (3 : ℕ))
+  ≤ ∑ j in (finset.range 3), x ^ j :=
+begin
+  norm_num [finset.sum],
+  rw [add_assoc, add_comm (x + 1) (x ^ 3 * 4 / 18), ← add_assoc, add_le_add_iff_right,
+      ← add_le_add_iff_left (-(x^2/2)), ← add_assoc, comm_ring.add_left_neg (x^2/2), zero_add,
+      neg_add_eq_sub, sub_half, sq, pow_succ, sq],
+  have i1 : x * 4 / 18 ≤ 1 / 2 := by linarith,
+  have i2 : 0 ≤ x * 4 / 18 := by linarith,
+  have i3 := mul_le_mul h1 h1 (le_refl (0 : ℝ)) h1,
```

Comment:
```suggestion
  have i3 := mul_le_mul h1 h1 le_rfl h1,
```

Just to note that this exists (and happens to work here, but could've not - I had to test it before recomemnding it). lemmas that end in `refl` usually have the argument that's reflexive specified, and `rfl` have it imp,icit

Reply:
for example, it works in line 1542, but I as expecting it to completely break the proof if I did it!